src/userscores/view/exportscoremodel.cpp

```diff
@@ -0,0 +1,363 @@
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "exportscoremodel.h"
+
+#include "log.h"
+#include "translation.h"
+
+#include "uicomponents/view/itemmultiselectionmodel.h"
+
+using namespace mu;
+using namespace mu::userscores;
+using namespace mu::notation;
+using namespace mu::uicomponents;
+using namespace mu::framework;
+
+ExportScoreModel::ExportScoreModel(QObject* parent)
+    : QAbstractListModel(parent), m_selectionModel(new ItemMultiSelectionModel(this))
+    , m_exportPath(getDefaultExportPath())
+    , m_currentConflictPolicy(FileConflictPolicy::Replace)
+{
+    connect(m_selectionModel, &ItemMultiSelectionModel::selectionChanged, this, &ExportScoreModel::selectionChanged);
+}
+
+void ExportScoreModel::load()
+{
+    beginResetModel();
+
+    IMasterNotationPtr masterNotation = this->masterNotation();
+    if (!masterNotation) {
+        endResetModel();
+        return;
+    }
+
+    m_notations << masterNotation->notation();
+    for (IExcerptNotationPtr excerpt : masterNotation->excerpts().val) {
+        m_notations << excerpt->notation();
+    }
+
+    endResetModel();
+}
+
+QVariant ExportScoreModel::data(const QModelIndex& index, int role) const
+{
+    if (!index.isValid()) {
+        return QVariant();
+    }
+
+    INotationPtr notation = m_notations[index.row()];
+
+    switch (role) {
+    case RoleTitle:
+        return notation->metaInfo().title;
+    case RoleIsSelected:
+        return m_selectionModel->isSelected(index);
+    case RoleIsMain:
+        return isMainNotation(notation);
+    }
+
+    return QVariant();
+}
+
+int ExportScoreModel::rowCount(const QModelIndex&) const
+{
+    return m_notations.size();
+}
+
+QHash<int, QByteArray> ExportScoreModel::roleNames() const
+{
+    static const QHash<int, QByteArray> roles {
+        { RoleTitle, "title" },
+        { RoleIsSelected, "isSelected" },
+        { RoleIsMain, "isMain" }
+    };
+
+    return roles;
+}
+
+void ExportScoreModel::toggleSelection(int scoreIndex)
+{
+    if (!isNotationIndexValid(scoreIndex)) {
+        return;
+    }
+
+    QModelIndex modelIndex = index(scoreIndex);
+    m_selectionModel->QItemSelectionModel::select(modelIndex, QItemSelectionModel::Toggle);
```

Comment:
ItemMultiSelectionModel's default behaviour is to select one item at a time unless ctrl/shift is held. Is that the right way to go here considering that each row has a checkbox and the user would expect toggling behaviour?

Reply:
We can ask @Tantacrul for help :) But we use this logic in all other places (see PartsDialog for example). Maybe you should just remove checkboxes because you already use highlighting for selected items?

Reply:
My first thought was to remove the checkboxes too.  But i wanted the checkboxes and the toggling behaviour because i expected most export operations to have multiple selections and having checkboxes would make that easier. I believe this question is best answered by @Tantacrul

Reply:
We've created a full design for this. @sidharth-anand - do you have access to the Figma files? If not, I can help you. @cbjeukendrup has access and can also advise.

Reply:
![image](https://user-images.githubusercontent.com/21022311/110792169-f72d0880-826a-11eb-9a91-591c8003b605.png)


For reference, this is how it should look. All pages / parts can be found (along with the exact colours and spacing) in the Figma file.

Reply:
@Tantacrul I'm in contact with Sidharth, we are currently working together to get the design implemented!